File: 240423/데이터센터의 온도 조정 2/adjusting-the-temperature-of-the-data-center-2.py
N, C, G, H = map(int, input().split())
degrees = [tuple(map(int, input().split())) for _ in range(N)]
Ta, Tb = list(list(zip(*degrees))[0]), list(list(zip(*degrees))[1])

max_time = 1000
max_works = 0
def get_works(l, h, t):
    if t < l:
        return C
    elif l <= t <= h:
        return G
    else:
        return H

def get_degrees(t):
    total = 0
    for ta, tb in zip(Ta, Tb):
        total += get_works(ta, tb, t)

    return total

for time in range(max_time + 1):
    max_works = max(max_works, get_degrees(time))

print(max_works)

File: 260313/데이터센터의 온도 조정 2/adjusting-the-temperature-of-the-data-center-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, C, G, H;
    cin >> N >> C >> G >> H;

    vector<pair<int,int>> degrees(N);
    for (int i = 0; i < N; i++) 
    {
        int a, b;
        cin >> a >> b;
        degrees[i] = {a, b};
    }

    auto get_works = [&](int l, int h, int t) -> int 
    {
        if (t < l) return C;
        if (l <= t && t <= h) return G;
        return H;
    };

    int max_works = 0;
    for (int t = 0; t <= 1000; t++) 
    {
        int total = 0;
        for (auto &p : degrees) {
            total += get_works(p.first, p.second, t);
        }
        max_works = max(max_works, total);
    }

    cout << max_works << "\n";
    return 0;
}